File: examples/tests/example-readings-axum-component-test/playbooks/calibration_default.rs
use arena_http::{ok_json, ManagedHttpPlaybook};
use serde_json::json;

use super::ids::{calibration_default_id, calibration_validate_path};

pub fn calibration_default_playbook(calibration_id: impl Into<String>) -> ManagedHttpPlaybook {
    ManagedHttpPlaybook::new(calibration_default_id(), calibration_id, |pb| {
        pb.post(calibration_validate_path())
            .will_return(ok_json(json!({ "valid": true })))
            .expect_called_at_least(1)
            .into_playbook()
    })
}

File: arena-pytest/tests/playbooks/calibration_default.py
from arena_pytest import HttpPlaybookBuilder, ManagedHttpPlaybook, ok_json

from readings_arena_config import (
    CALIBRATION_VALIDATE_PATH,
    PLAYBOOK_CALIBRATION_DEFAULT,
)


class CalibrationDefaultPlaybook(ManagedHttpPlaybook):
    def __init__(self, dependency_identifier: str):
        super().__init__(
            identifier=PLAYBOOK_CALIBRATION_DEFAULT,
            dependency_identifier=dependency_identifier,
            builder=(
                HttpPlaybookBuilder(dependency_identifier)
                .post(CALIBRATION_VALIDATE_PATH)
                .will_return(ok_json({"valid": True}))
                .expect_called_at_least(1)
            ),
        )
